bftengine/src/bftengine/ReadOnlyReplica.cpp

```diff
@@ -40,7 +40,9 @@ ReadOnlyReplica::ReadOnlyReplica(const ReplicaConfig &config,
                   metrics_.RegisterCounter("receivedInvalidMsgs"),
                   metrics_.RegisterGauge("lastExecutedSeqNum", lastExecutedSeqNum)} {
   LOG_INFO(GL, "");
-  bftEngine::ControlStateManager::instance().setRemoveMetadataFunc([&]() { stateTransfer->setEraseMetadataFlag(); });
+  bftEngine::ControlStateManager::instance().setRemoveMetadataFunc([&](bool include_st) {
```

Comment:
Consider moving this to the `ReplicaBse` and  `ReplicaForStateTransfer` correspondingly

Reply:
ReplicaImp also derived from ReplicaForStateTransfer and it sets a different method (that includes the bft metadata)

Reply:
So split st metadata to be treated at `ReplicaForStateTransfer` and bft mdt in ReplicaImp

Reply:
fixed